tests/state/test_state_api.py

```diff
@@ -180,3 +181,24 @@ def do_job():
             {"name": "do_work", "priority": 0, "queue_name": "default"},
             {"name": "do_job", "priority": 10, "queue_name": "foo"},
         ]
+
+    def test_filter_messages_id(self, stub_broker, api_client, state_middleware):
+        random_states = []
+        for i in range(10):
```

Comment:
no need for random test to only get the first each time. maybe insert only the first. make the id longer and only search a part of it.

Add a test for offset and page size.